packages/core/src/tracing/instrumentation/protocols/nativeFetch.js

```diff
@@ -0,0 +1,304 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+'use strict';
+
+const cls = require('../../cls');
+const constants = require('../../constants');
+const httpCommon = require('./_http');
+const tracingUtil = require('../../tracingUtil');
+const { sanitizeUrl, splitAndFilter } = require('../../../util/url');
+
+const originalFetch = global.fetch;
+
+let extraHttpHeadersToCapture;
+let isActive = false;
+
+exports.init = function init(config) {
+  if (originalFetch == null) {
+    // Do nothing in Node.js versions that do not support native fetch.
+    return;
+  }
+
+  instrument();
+  extraHttpHeadersToCapture = config.tracing.http.extraHttpHeadersToCapture;
+};
+
+exports.updateConfig = function updateConfig(config) {
+  extraHttpHeadersToCapture = config.tracing.http.extraHttpHeadersToCapture;
+};
+
+exports.activate = function activate(extraConfig) {
+  if (originalFetch == null) {
+    // Do nothing in Node.js versions that do not support native fetch.
+    return;
+  }
+
+  if (
+    extraConfig &&
+    extraConfig.tracing &&
+    extraConfig.tracing.http &&
+    Array.isArray(extraConfig.tracing.http.extraHttpHeadersToCapture)
+  ) {
+    extraHttpHeadersToCapture = extraConfig.tracing.http.extraHttpHeadersToCapture;
+  }
+  isActive = true;
+};
+
+exports.deactivate = function deactivate() {
+  isActive = false;
+};
+
+function instrument() {
+  global.fetch = function fetch() {
+    // eslint-disable-next-line no-unused-vars
+    let w3cTraceContext = cls.getW3cTraceContext();
+
+    const skipTracingResult = cls.skipExitTracing({ isActive, extendedResponse: true, skipParentSpanCheck: true });
+    const parentSpan = cls.getCurrentSpan() || cls.getReducedSpan();
```

Comment:
help: I do understand the concept of reduced spans, but I currently struggle why we need this for http clients. Makes only sense to me for graphql subscriptions. Can we leave a comment?

Reply:
To be honest, I started the native fetch instrumentation by taking bits and pieces from the existing `httpClient` instrumentation, and that instrumentation also [falls back to the reduced span](https://github.com/instana/nodejs/blob/13c4ff3143c8ad6929d2b66414ba8420d75109a3/packages/core/src/tracing/instrumentation/protocols/httpClient.js#L150) if there is no current span.

This had been added there with [this commit](https://github.com/instana/nodejs/commit/55db775df23d0d28d342552ff7e8dfcb4051a3b6):

> feat(tracing): support tracing of deferred HTTP exits
> Also capture HTTP exits if the entry that triggered it has already
finished. 

Which means: It can happen that processing an HTTP entry has already finished when an outgoing HTTP request is made. I do not remember the concrete use case that prompted implementing it like this. I do remember that it was a real world use case from a customer, though, which got fixed by adding this fallback in `httpClient.js`.

Reply:
Discussed f2f, will add an additional comment that explains this a bit more.